pkg/crc/machine/bundle/metadata.go

```diff
@@ -281,3 +282,23 @@ func DownloadDefault(preset preset.Preset) error {
 
 	return nil
 }
+
+func Download(preset crcPreset.Preset, bundlePath string, isDefault bool) (string, error) {
+	if isDefault {
+		if preset == crcPreset.OpenShift {
+			return bundlePath, DownloadDefault(preset)
+		}
+		return bundlePath, image.PullBundle(preset, "")
+	}
```

Comment:
The `isDefault` code was moved here, but this still duplicates the code in the `switch` below, so this still has the same issue as the previous code.

Reply:
@cfergeau Yes, May be this is something we can take in a follow up PR?

Reply:
I did some work on this in https://github.com/cfergeau/crc/commits/issue_3277 to try to remove at least the `isDefault` parameter which seems redundant.
Not sure there's a lot more we can rework, maybe some small `downloaderFromURI()` helper which would return opaque types with just a `Download()` method and a `HasIntegrityCheck()`, and these opaque types would wrap either a DownloadInfo for default bundles, or `PullBundle` for container images, or `Download` for http URLs.
That might also be a good opportunity to push the default container registry URLs to `bundleLocations` rather than maintaining it in a different part of the code.

Reply:
should download not just pull the bundle? or perhaps pull the verification of the gpg key out and place this more logically as part of last step here?